llmfoundry/utils/builders.py

```diff
@@ -497,8 +497,16 @@ def _validate_cfg(icl_cfg: DictConfig):
             icl_cfg.batch_size = default_batch_size
         if 'pass_at_k' not in icl_cfg:
             icl_cfg.pass_at_k = 1
-        if 'num_beams' not in icl_cfg:
-            icl_cfg.num_beams = 20
+        if 'generations_per_sample' not in icl_cfg:
+            if 'num_beams' in icl_cfg:
+                icl_cfg.generations_per_sample = icl.num_beams
+                warnings.warn(
+                    ('num_beams is deprecated and will be removed in a future release.'
+                     f'Please use generations_per_sample: {icl.num_beams}'),
+                    DeprecationWarning,
```

Comment:
We're deprecating num_beams entirely in favor of passing generation keyword args in w/ `generation_kwargs`, mind updating the warning to say so?

Reply:
(also icl.num_beams here)

Reply:
lol thanks, fixed the icl.num_beams

Reply:
I think something more like this:
```
'num_beams' is deprecated and will be removed in a future release. '
 'To specify keyword arguments for `model.generate()`, please use '
 '`generation_kwargs` with your corresponding values.' 
```
The idea is we don't want them to use generations_per_sample or num_beams - we want them to put any and all generation_kwargs in that dictionary, like this:
```
label: winograd
dataset_uri: eval/local_data/language_understanding/winograd_wsc.jsonl # ADD YOUR OWN DATASET URI
num_fewshot: [0]
icl_task_type: schema
generation_kwargs:
  num_beams: 9001
  temperateure: .9
```
(these are not real params btw)

We should also have a deprecation warning for `generations_per_sample` even though we're literally introducing it right now XD

also these strings are throwing a syntax error lol

Reply:
According to the PR [Make CodeEval respect device_eval_batch_size](https://github.com/mosaicml/composer/pull/2969), `generations_per_sample` does not go in `generation_kwargs` -- it is it's own parameter. Thus we do not want to put it in `generation_kwargs`.